src/lib.rs

```diff
@@ -15,18 +15,58 @@ pub enum CharacterSet {
     UrlSafe
 }
 
+#[derive(Clone, Copy, Debug)]
+pub enum LineEnding {
+    LF,
+    CRLF,
+}
+
 /// Contains configuration parameters for base64 encoding
 #[derive(Clone, Copy, Debug)]
 pub struct Config {
     /// Character set to use
     pub char_set: CharacterSet,
     /// True to pad output with `=` characters
     pub pad: bool,
+    /// Remove whitespace before decoding, at the cost of an allocation
+    pub strip_whitespace: bool,
+    /// Characters per line, None (or Some(0), but less "proper") for no linebreaks
+    pub line_size: Option<usize>,
+    /// Unix or Windows line endings, ignored if above None/Some(0)
+    pub line_ending: LineEnding,
```

Comment:
I think it might be nice to have the type system provide more guidance about the line wrap stuff, perhaps by having a dedicated `LineWrap` struct. There's basically an implicit `NoWrap` line wrap style that is applied by having line_size be 0 or `None`. Leaking that detail to the outside world means we might end up in a compatibility pickle as we make more changes to Config. If we hid that detail from the world, we could have more freedom to change that later, and also avoid user confusion about the sometimes-ignored `line_ending` member. So, something like:

```
struct LineWrap {
    line_size: Option<usize>,
    line_ending: LineEnding
}

impl struct LineWrap {
    pub fn no_wrap() -> LineWrap {
        LineWrap {line_size: None, line_ending: LF}
    }

    pub fn wrap(line_size: usize, line_ending: LineEnding) -> LineWrap {
        // check line_size > 0
        ...
    }
}
```

Now we can hide the details of how `line_ending` is applied, and the only two valid use cases are explicitly labelled with those methods.

A second thought... This `Config` struct is getting a little beefy. `char_set` is meaningful for both encoding and decoding, but the others are only used for either encoding or decoding, not both.

WDYT about a `EncodeConfig` with `char_set`, `pad`, and `line_wrap`, and a `DecodeConfig` with `char-set` and `strip_whitespace`?